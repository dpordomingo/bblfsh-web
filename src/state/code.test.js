import {
  reducer,
  initialState,
  SET,
  PARSE,
  PARSE_FAILED,
  SET_AST,
  CHANGE,
  MARK,
} from './code';
import {
  NODE_EXPAND,
  NODE_TOGGLE,
  NODE_HIGHLIGHT,
  NODE_UNHIGHLIGHT,
} from './ast';

describe('code/reducer', () => {
  it('SET', () => {
    expect(
      reducer(initialState, {
        type: SET,
        filename: 'file.py',
        code: 'foo = 1',
        ast: { InternalType: 'file' },
      })
    ).toMatchSnapshot();
  });

  it('PARSE', () => {
    expect(
      reducer(
        {
          ...initialState,
          ast: { 1: { id: 1 } },
          posIndex: 'index',
          parsing: false,
        },
        {
          type: PARSE,
        }
      )
    ).toMatchSnapshot();
  });

  it('PARSE_FAILED', () => {
    expect(
      reducer({ ...initialState, parsing: true }, { type: PARSE_FAILED })
    ).toMatchSnapshot();
  });

  it('SET_AST', () => {
    expect(
      reducer(initialState, { type: SET_AST, ast: { 1: { id: 1 } } })
    ).toMatchSnapshot();
  });

  it('CHANGE', () => {
    expect(
      reducer(
        { ...initialState, code: 'foo = 1' },
        { type: CHANGE, code: 'bar = 1' }
      )
    ).toMatchSnapshot();
  });

  it('MARK', () => {
    expect(
      reducer(initialState, { type: MARK, range: { from: 1, to: 2 } })
    ).toMatchSnapshot();
  });

  const initialWithNode = {
    ...initialState,
    ast: {
      1: {
        id: 1,
        expanded: false,
        higlighted: false,
      },
    },
  };

  it('NODE_EXPAND', () => {
    expect(
      reducer(initialWithNode, { type: NODE_EXPAND, nodeId: 1 })
    ).toMatchSnapshot();
  });

  it('NODE_TOGGLE', () => {
    expect(
      reducer(initialWithNode, { type: NODE_TOGGLE, nodeId: 1 })
    ).toMatchSnapshot();

    expect(
      reducer(
        {
          ...initialWithNode,
          ast: {
            ...initialWithNode.ast,
            1: {
              ...initialWithNode.ast[1],
              expanded: true,
            },
          },
        },
        { type: NODE_TOGGLE, nodeId: 1 }
      )
    ).toMatchSnapshot();
  });

  it('NODE_HIGHLIGHT', () => {
    expect(
      reducer(initialWithNode, { type: NODE_HIGHLIGHT, nodeId: 1 })
    ).toMatchSnapshot();
  });

  it('NODE_UNHIGHLIGHT', () => {
    expect(
      reducer(
        {
          ...initialWithNode,
          ast: {
            ...initialWithNode.ast,
            1: {
              ...initialWithNode.ast[1],
              higlighted: true,
            },
          },
        },
        { type: NODE_UNHIGHLIGHT, nodeId: 1 }
      )
    ).toMatchSnapshot();
  });
});
