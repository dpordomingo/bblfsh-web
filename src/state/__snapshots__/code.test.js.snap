// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code/reducer CHANGE 1`] = `
Object {
  "ast": Object {},
  "code": "bar = 1",
  "dirty": true,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer MARK 1`] = `
Object {
  "ast": Object {},
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": Object {
    "from": 1,
    "to": 2,
  },
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer NODE_EXPAND 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "expanded": true,
      "higlighted": false,
      "id": 1,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer NODE_HIGHLIGHT 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "expanded": false,
      "highlighted": true,
      "higlighted": false,
      "id": 1,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer NODE_TOGGLE 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "expanded": true,
      "higlighted": false,
      "id": 1,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer NODE_TOGGLE 2`] = `
Object {
  "ast": Object {
    "1": Object {
      "expanded": false,
      "higlighted": false,
      "id": 1,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer NODE_UNHIGHLIGHT 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "expanded": false,
      "highlighted": false,
      "higlighted": true,
      "id": 1,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer PARSE 1`] = `
Object {
  "ast": Object {},
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": true,
  "posIndex": null,
}
`;

exports[`code/reducer PARSE_FAILED 1`] = `
Object {
  "ast": Object {},
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": null,
}
`;

exports[`code/reducer SET 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "InternalType": "file",
      "expanded": true,
      "higlighted": false,
      "id": 1,
      "parentId": undefined,
    },
  },
  "code": "foo = 1",
  "dirty": true,
  "filename": "file.py",
  "markRange": null,
  "parsing": false,
  "posIndex": NodeIndex {
    "index": Array [],
  },
}
`;

exports[`code/reducer SET_AST 1`] = `
Object {
  "ast": Object {
    "1": Object {
      "1": Object {
        "id": 1,
      },
      "expanded": true,
      "higlighted": false,
      "id": 1,
      "parentId": undefined,
    },
  },
  "code": null,
  "dirty": false,
  "filename": undefined,
  "markRange": null,
  "parsing": false,
  "posIndex": NodeIndex {
    "index": Array [],
  },
}
`;
